lib/Sema/CSSimplify.cpp

```diff
@@ -9468,7 +9468,20 @@ ConstraintSystem::simplifyKeyPathConstraint(
     case KeyPathExpr::Component::Kind::Invalid:
     case KeyPathExpr::Component::Kind::Identity:
       break;
-      
+
+    case KeyPathExpr::Component::Kind::CodeCompletion: {
+      // Declare the code completion component as unresolved during constraint
+      // generation because we don't have enough information yet to decide
+      // whether the key path should be a BGT KeyPath or a function type.
+      // During solving, always declare the component as resolved because we
+      // won't get any more information out of it in subsequent passes - its
+      // result is always a type variable with no constraints on the preceeding
+      // components.
+      if (flags.contains(TMF_GenerateConstraints)) {
```

Comment:
`flags` doesn't have any relation to constraint generation phase, you'd need to check `getPhase() == ConstraintSystemPhase::ConstraintGeneration` for that. `TMF_GenerateConstraints` is used in cases were it's okay to add new constraints  while solving current one e.g. simplification logic sometimes has to generate new identical constraint if current one can't be simplified, `matchTypes` has to add new constraints when it's matching two bound generic types in order to match generic generic arguments etc.